core/src/ledger/governance/utils.rs

```diff
@@ -196,12 +218,25 @@ impl TallyVote {
         }
     }
 
+    /// Check if a vote is nay
+    pub fn is_nay(&self) -> bool {
+        match self {
+            TallyVote::OnChain(vote) => vote.is_nay(),
+            TallyVote::Offline(vote) => vote.is_nay(),
+        }
+    }
+
     /// Check if two votes are equal
     pub fn is_same_side(&self, other: &TallyVote) -> bool {
-        let both_yay = self.is_yay() && other.is_yay();
-        let both_nay = !self.is_yay() && !other.is_yay();
-
-        both_yay || !both_nay
+        match (self, other) {
+            (TallyVote::OnChain(vote), TallyVote::OnChain(other_vote)) => {
+                vote.is_same_side(other_vote)
+            }
+            (TallyVote::Offline(vote), TallyVote::Offline(other_vote)) => {
+                vote.is_same_side(other_vote)
+            }
+            _ => false,
```

Comment:
when would this case happen? shouldn't this be impossible?

Reply:
Never, it's just the the method is implemented on the `TallyVote` enum which contains two variants, one for the on-chain votes and the other one for off-chain votes and we still need the pattern matching to be exhaustive, so we need to cover also the case in which the two instances are actually two different variants. We can return an error instead of `false` to make it more clear

Reply:
wherever a case isn't expected to happen, we should return an error